src/state/wallets/walletsStore.ts

```diff
@@ -522,110 +528,111 @@ export const useWalletsStore = createRainbowStore<WalletsState>(
 
     checkKeychainIntegrity: async () => {
       try {
+        const startTime = Date.now();
```

Comment:
This is a complete rewrite of `checkKeychainIntegrity` with our current requirements.